### PR TITLE
High memory workers in Combiner stage of Large Cohorts

### PR DESCRIPTION
**Pull request by michael-harper**
The Combiner stage was recently taken off of Dataproc (https://github.com/populationgenomics/production-pipelines/pull/621). 

Outputs from this stage when not using Dataproc were tested for consistency, however only on test-level data. Running on main has produced OOM issues that may be remedied by using high memory workers. Currently, there is a config parameter `highmem_workers` available but is only retrieved when setting up a [dataproc_job](https://github.com/populationgenomics/production-pipelines/blob/a2cdfa4de9d6b178fd285eaca0114ddec1bddb26/cpg_workflows/large_cohort/dataproc_utils.py#L103). This PR retrieves the same parameter for use in the `query_command` function that consequently uses the parameter as a batch override via `init_batch`.

### CODE REVIEW

**Comment by hopedisastro** [comment]
prev. discussion on QoB OOMing out: https://centrepopgen.slack.com/archives/C030X7WGFCL/p1707187236506529

**Comment by hopedisastro** [comment]
Alternative solution - increase driver cores in init_batch(): 
https://github.com/populationgenomics/tob-wgs/pull/102
https://centrepopgen.slack.com/archives/C018KFBCR1C/p1658811940454569?thread_ts=1658208231.938539&cid=C018KFBCR1C

File: cpg_workflows/stages/large_cohort.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
 
         j = get_batch().new_job('Combiner', (self.get_job_attrs() or {}) | {'tool': 'hail query'})
 
+        init_batch_args = {'worker_memory': 'highmem'} if get_config()['workflow'].get('highmem_works') else None
```
Comment (MattWellie) [inline cpg_workflows/stages/large_cohort.py:34]: do you want to add `highmem_works` to the default large_cohort config?

**Review by MattWellie — APPROVED** [review]

File: cpg_workflows/stages/large_cohort.py (hunk 1 above)
Comment (vivbak) [reply cpg_workflows/stages/large_cohort.py:34]: ^ second this

**Review by vivbak — APPROVED** [review]

File: cpg_workflows/stages/large_cohort.py (hunk 1 above)
Comment (michael-harper) [reply cpg_workflows/stages/large_cohort.py:34]: Added!

**Pull request merged**